htlcswitch/link.go

```diff
@@ -1529,10 +1548,36 @@ func (l *channelLink) handleDownStreamPkt(pkt *htlcPacket, isReProcess bool) {
 		htlc.ChanID = l.ChanID()
 		htlc.ID = pkt.incomingHTLCID
 
-		// Finally, we send the HTLC message to the peer which
-		// initially created the HTLC.
+		// We send the HTLC message to the peer which initially created
+		// the HTLC.
 		l.cfg.Peer.SendMessage(false, htlc)
 		isSettle = true
+
+		// Finally, we notify the failure.
+		key := newHtlcKey(pkt)
+		info := HtlcInfo{
+			IncomingTimeLock: pkt.incomingTimeout,
+			OutgoingTimeLock: pkt.outgoingTimeout,
+			IncomingAmt:      pkt.incomingAmount,
+			OutgoingAmt:      pkt.amount,
+			HtlcEventType:    getEventType(pkt),
```

Comment:
Payment hash is not immediately available here, unsure of how critical this is given that we identify htlcs by circuit key? 

The only place I can see this being a problem is when you subscribe to events after the forward and before the failure, which is the case where we would not know the hash this is associated with.

Reply:
Shouldn't we even discourage users of the htlc notifier of using the hash to relate events? Maybe none of the 'outcome' events should include a hash.

Reply:
> Maybe none of the 'outcome' events should include a hash.

I think this is a  good idea. I will move hash into forwarding events, and take it out of settles/fails.